File: src/config/events.ts
import { CHART_PALETTE } from '@/config/theme';

// Hardcoded from notebook section 1.2: event tags and macro context.
// Users can add events via the UI; these are the built-in defaults.

export interface EventDef {
  name: string;
  date: string;
}

export const EVENTS: EventDef[] = [
  { name: '1973 Oil Embargo', date: '1973-10-17' },
  { name: '1990 Gulf War', date: '1990-08-02' },
  { name: '1991 Kuwait Oil Fires', date: '1991-01-16' },
  { name: '1998 Desert Fox', date: '1998-12-16' },
  { name: '2001 Afghanistan (OEF)', date: '2001-10-07' },
  { name: '2003 SARS', date: '2003-03-12' },
  { name: '2003 Iraq War', date: '2003-03-20' },
  { name: '2011 Libya', date: '2011-03-19' },
  { name: '2014 ISIS/Mosul', date: '2014-06-10' },
  { name: '2017 Syria Strikes', date: '2017-04-07' },
  { name: 'COVID-19', date: '2020-03-01' },
  { name: '2022 Russia-Ukraine', date: '2022-02-24' },
  { name: '2023 Red Sea Crisis', date: '2023-12-19' },
];

export const ALL_TAGS = [
  'energy_shock',
  'military_conflict',
  'shipping_disruption',
  'sanctions',
  'pandemic',
] as const;

export type EventTag = typeof ALL_TAGS[number];

export const EVENT_TAGS: Record<string, Set<string>> = {
  '1973 Oil Embargo': new Set(['energy_shock', 'sanctions']),
  '1990 Gulf War': new Set(['military_conflict', 'energy_shock']),
  '1991 Kuwait Oil Fires': new Set(['military_conflict', 'energy_shock']),
  '1998 Desert Fox': new Set(['military_conflict']),
  '2001 Afghanistan (OEF)': new Set(['military_conflict']),
  '2003 SARS': new Set(['pandemic']),
  '2003 Iraq War': new Set(['military_conflict', 'energy_shock']),
  '2011 Libya': new Set(['military_conflict', 'energy_shock']),
  '2014 ISIS/Mosul': new Set(['military_conflict', 'energy_shock']),
  '2017 Syria Strikes': new Set(['military_conflict']),
  'COVID-19': new Set(['pandemic']),
  '2022 Russia-Ukraine': new Set(['military_conflict', 'energy_shock', 'sanctions']),
  '2023 Red Sea Crisis': new Set(['shipping_disruption', 'military_conflict']),
};

export interface MacroContext {
  trigger: number;
  cpi: 'high' | 'mid' | 'low';
  fed: 'hiking' | 'cutting' | 'hold';
}

export const MACRO_CONTEXT: Record<string, MacroContext> = {
  '1973 Oil Embargo': { trigger: 4, cpi: 'high', fed: 'hiking' },
  '1990 Gulf War': { trigger: 17, cpi: 'high', fed: 'cutting' },
  '1991 Kuwait Oil Fires': { trigger: 25, cpi: 'high', fed: 'cutting' },
  '1998 Desert Fox': { trigger: 11, cpi: 'low', fed: 'hold' },
  '2001 Afghanistan (OEF)': { trigger: 22, cpi: 'low', fed: 'cutting' },
  '2003 SARS': { trigger: 35, cpi: 'low', fed: 'cutting' },
  '2003 Iraq War': { trigger: 37, cpi: 'low', fed: 'cutting' },
  '2011 Libya': { trigger: 85, cpi: 'mid', fed: 'hold' },
  '2014 ISIS/Mosul': { trigger: 104, cpi: 'low', fed: 'hold' },
  '2017 Syria Strikes': { trigger: 53, cpi: 'mid', fed: 'hiking' },
  'COVID-19': { trigger: 54, cpi: 'low', fed: 'cutting' },
  '2022 Russia-Ukraine': { trigger: 91, cpi: 'high', fed: 'hiking' },
  '2023 Red Sea Crisis': { trigger: 73, cpi: 'mid', fed: 'hold' },
};

export const EVENT_COLORS = CHART_PALETTE;

File: config/events.py
from __future__ import annotations

from engine.models import EventDef, MacroContext


EVENTS = [
    EventDef("1973 Oil Embargo", "1973-10-17"),
    EventDef("1990 Gulf War", "1990-08-02"),
    EventDef("1991 Kuwait Oil Fires", "1991-01-16"),
    EventDef("1998 Desert Fox", "1998-12-16"),
    EventDef("2001 Afghanistan (OEF)", "2001-10-07"),
    EventDef("2003 SARS", "2003-03-12"),
    EventDef("2003 Iraq War", "2003-03-20"),
    EventDef("2011 Libya", "2011-03-19"),
    EventDef("2014 ISIS/Mosul", "2014-06-10"),
    EventDef("2017 Syria Strikes", "2017-04-07"),
    EventDef("COVID-19", "2020-03-01"),
    EventDef("2022 Russia-Ukraine", "2022-02-24"),
    EventDef("2023 Red Sea Crisis", "2023-12-19"),
]

ALL_TAGS = [
    "energy_shock",
    "military_conflict",
    "shipping_disruption",
    "sanctions",
    "pandemic",
]

EVENT_TAGS = {
    "1973 Oil Embargo": {"energy_shock", "sanctions"},
    "1990 Gulf War": {"military_conflict", "energy_shock"},
    "1991 Kuwait Oil Fires": {"military_conflict", "energy_shock"},
    "1998 Desert Fox": {"military_conflict"},
    "2001 Afghanistan (OEF)": {"military_conflict"},
    "2003 SARS": {"pandemic"},
    "2003 Iraq War": {"military_conflict", "energy_shock"},
    "2011 Libya": {"military_conflict", "energy_shock"},
    "2014 ISIS/Mosul": {"military_conflict", "energy_shock"},
    "2017 Syria Strikes": {"military_conflict"},
    "COVID-19": {"pandemic"},
    "2022 Russia-Ukraine": {"military_conflict", "energy_shock", "sanctions"},
    "2023 Red Sea Crisis": {"shipping_disruption", "military_conflict"},
}

MACRO_CONTEXT = {
    "1973 Oil Embargo": MacroContext(4, "high", "hiking"),
    "1990 Gulf War": MacroContext(17, "high", "cutting"),
    "1991 Kuwait Oil Fires": MacroContext(25, "high", "cutting"),
    "1998 Desert Fox": MacroContext(11, "low", "hold"),
    "2001 Afghanistan (OEF)": MacroContext(22, "low", "cutting"),
    "2003 SARS": MacroContext(35, "low", "cutting"),
    "2003 Iraq War": MacroContext(37, "low", "cutting"),
    "2011 Libya": MacroContext(85, "mid", "hold"),
    "2014 ISIS/Mosul": MacroContext(104, "low", "hold"),
    "2017 Syria Strikes": MacroContext(53, "mid", "hiking"),
    "COVID-19": MacroContext(54, "low", "cutting"),
    "2022 Russia-Ukraine": MacroContext(91, "high", "hiking"),
    "2023 Red Sea Crisis": MacroContext(73, "mid", "hold"),
}
